parts/xmem_nozzle.go

```diff
@@ -3695,6 +3708,9 @@ func (xmem *XmemNozzle) receiveResponse(finch chan bool, waitGrp *sync.WaitGroup
 					CLogWaitTime:         cLogWaitTime,
 					CLogError:            cLogErr,
 				}
+				if wrappedReq.OrigSrcVB != nil {
+					additionalInfo.SetOrigSrcVB(*wrappedReq.OrigSrcVB)
+				}
 				xmem.RaiseEvent(common.NewEvent(common.DataSent, nil, xmem, nil, additionalInfo))
```

Comment:
DerivedData should be passed here.

I ran mobile tests (runHlv 0 and 0a with simulator) and it panics because the DerivedData is expected.

cc: @darshan-na

Reply:
This is for stats:
```
DOCS_SENT_WITH_SUBDOC_SET    = base.DocsSentWithSubdocSet
DOCS_SENT_WITH_SUBDOC_DELETE = base.DocsSentWithSubdocDelete
```

Reply:
To run the tests with simulator:
1. clone https://github.com/couchbaselabs/MobileImportSim/tree/main
2. create the binary - `make clean && make deps && make`
3. Run the tests in goxdcr/tools/testScripts - `./runHlvTests.sh --run=0 --mobileImportSimDir="/Users/sumukhbhat2701/work/MobileImportSim"` & `./runHlvTests.sh --run=0a --mobileImportSimDir="/Users/sumukhbhat2701/work/MobileImportSim"` => Replace /Users/sumukhbhat2701/work/MobileImportSim with the directory created in (2)